src/T2DMap.cpp

```diff
@@ -215,6 +215,21 @@ QColor T2DMap::getColor(int id)
         break;
     default: //user defined room color
         if (!mpMap->customEnvColors.contains(env)) {
+            if (16 < env < 232)
+            {
+                int base = env - 16;
+                int r = base / 36;
+                int g = (base - (r * 36)) / 6;
+                int b = (base - (r * 36)) - (g * 6);
+
+                r = r * 51;
+                g = g * 51;
+                b = b * 51;
+                color = QColor(r, g, b, 255);
+            } else if (231 < env < 256) {
+                int k = ((env - 232) * 10) + 8;
```

Comment:
You might consider the straight lookup table I use elsewhere (see `TBuffer::translateToPlanText(...)`):
```C
                                    // black + 23 tone grayscale from dark to light gray
                                    // Similar to RGB case the multiplier is a bit off
                                    // we have been using 10 but 23 x 10 = 230
                                    // whereas 23 should map to 255, this requires
                                    // a non-integer multiplier, instead of multiplying
                                    // and rounding we, for speed, can use a look-up table:
                                    int value;
                                    // clang-format off
                                    switch (tag) {
                                        case 232:   value =   0; break; //   0.000
                                        case 233:   value =  11; break; //  11.087
                                        case 234:   value =  22; break; //  22.174
                                        case 235:   value =  33; break; //  33.261
                                        case 236:   value =  44; break; //  44.348
                                        case 237:   value =  55; break; //  55.435
                                        case 238:   value =  67; break; //  66.522
                                        case 239:   value =  78; break; //  77.609
                                        case 240:   value =  89; break; //  88.696
                                        case 241:   value = 100; break; //  99.783
                                        case 242:   value = 111; break; // 110.870
                                        case 243:   value = 122; break; // 121.957
                                        case 244:   value = 133; break; // 133.043
                                        case 245:   value = 144; break; // 144.130
                                        case 246:   value = 155; break; // 155.217
                                        case 247:   value = 166; break; // 166.304
                                        case 248:   value = 177; break; // 177.391
                                        case 249:   value = 188; break; // 188.478
                                        case 250:   value = 200; break; // 199.565
                                        case 251:   value = 211; break; // 210.652
                                        case 252:   value = 222; break; // 221.739
                                        case 253:   value = 233; break; // 232.826
                                        case 254:   value = 244; break; // 243.913
                                        case 255:   value = 255; break; // 255.000
                                    }
                                    // clang-format on
```
substituting `env` for `tag` and `quint8 k` (1 byte rather than 4 for an `int k`) for `value`...

Reply:
I'd rather stick with the actual calculation, as these values aren't technically correct. 232 and 255 aren't true black/white (since those colours are already supported elsewhere), they're just the last steps between black and white. The differences are minute enough not to matter in almost any case, but 🤷‍♂️

Reply:
They do reflect the colours we've used elsewhere however.

Reply:
True, but that doesn't make them right :D

I can change it around if you'd really like.